snuba/subscriptions/executor_consumer.py

```diff
@@ -222,17 +224,28 @@ def submit(self, message: Message[KafkaPayload]) -> None:
 
         tick_upper_offset = task.task.tick_upper_offset
 
-        self.__queue.append(
-            (
-                message,
-                SubscriptionTaskResultFuture(
-                    task,
-                    self.__executor.submit(
-                        self.__execute_query, task, tick_upper_offset
+        # We need to sample queries to ClickHouse while this is being rolled out
+        # as we don't want to duplicate every subscription query
+        executor_sample_rate = cast(
+            float, state.get_config("executor_sample_rate", 0.0)
+        )
+        subscription_id = str(task.task.subscription.identifier)
+        should_execute = (
+            (crc32(subscription_id.encode("utf-8")) & 0xFFFFFFFF) / 2 ** 32
```

Comment:
Why the bitwise AND ? `& 0xFFFFFFFF` should not be needed.

Reply:
Probably not needed but it was suggested in the docs https://docs.python.org/3/library/zlib.html#zlib.crc32 to make it compatible across all platforms and python versions